### PR TITLE
Benchmark v0.4

### PR DESCRIPTION
**Pull request by ab93**
Benchmarking with respect to a subset of a public dataset.

### CODE REVIEW

File: benchmarks/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Benchmarks
+
+This section contains some benchmarking results of numalogic's algorithms on real as well
+synthetic data. Datasets here are publicly available from their respective repositories.
+
+Note that efforts have not really been made on hyperparameter tuning. This is just to give users an 
+idea on how each algorithm is suitable for different kinds of data, and shows how they can do
+their own benchmarking too.
+
+This is an ongoing process, and will add more benchmarking results in the near future.
```
Comment (qhuai) [inline benchmarks/README.md:10]: nitpick: This is an ongoing process, and `we` will ...

File: benchmarks/kpi/datamodule.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+import numpy as np
+import pandas as pd
+from sklearn.pipeline import make_pipeline
+from torch.utils.data import DataLoader
+
+from numalogic.tools.data import TimeseriesDataModule, StreamingDataset
+
+
+class KPIDataModule(TimeseriesDataModule):
```
Comment (qhuai) [inline benchmarks/kpi/datamodule.py:11]: Can you add some overall comments describing the KPIDataModule?

File: benchmarks/kpi/datamodule.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        self._train_labels = None
+        self._val_labels = None
+        self._test_labels = None
+
+        self.unique_kpis = None
+
+        self._kpi_df = self.get_kpi_df()
+
+    def _preprocess(self, df: pd.DataFrame):
```
Comment (qhuai) [inline benchmarks/kpi/datamodule.py:49]: What's the expected structure within df?

File: benchmarks/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+## Benchmarks
```
Comment (qhuai) [inline benchmarks/README.md:1]: Can you add steps so that another engineer can repeat the KPI results?

For example, the jupyter notebook assumes `DATA_FILE_PATH = "data/Finals_dataset/phase2_train.csv"`. Is there a separate step to download the data?

**Review by qhuai — COMMENTED** [review]
initial comments

File: benchmarks/kpi/datamodule.py (hunk 3 above)
Comment (ab93) [reply benchmarks/kpi/datamodule.py]: Added in the docstrings

File: benchmarks/README.md (hunk 4 above)
Comment (ab93) [reply benchmarks/README.md:1]: I mentioned this in the docstrings. Let me know if I should add it to the notebook as well.

File: benchmarks/kpi/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+## KPI Anomaly dataset
```
Comment (s0nicboOm) [inline benchmarks/kpi/README.md:1]: I think we will have to add this to copyright for our data: https://github.com/NetManAIOps/KPI-Anomaly-Detection/blob/master/LICENSE

File: benchmarks/kpi/results/kpi_idx_14/transformer/train.png
Comment (s0nicboOm) [inline benchmarks/kpi/results/kpi_idx_14/transformer/train.png:1]: I think we should add labels to the images. Like x axis is timestamp. (Maybe if we can have timestamp itself.)

File: benchmarks/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+This section contains some benchmarking results of numalogic's algorithms on real as well
+synthetic data. Datasets here are publicly available from their respective repositories.
+
+Note that efforts have not really been made on hyperparameter tuning. This is just to give users an 
+idea on how each algorithm is suitable for different kinds of data, and shows how they can do
```
Comment (nkoppisetty) [inline benchmarks/README.md:7]: Overall LGTM. But, I think there should be a more detailed Readme explaining things in simpler terms so that even non ML engineers could understand it.

File: benchmarks/kpi/README.md (hunk 5 above)
Comment (ab93) [reply benchmarks/kpi/README.md:1]: What do you mean by adding copyright? They have specified their license. And we are not adding their data in the repository. Just running the analysis on it.

File: benchmarks/kpi/results/kpi_idx_14/transformer/train.png
Comment (ab93) [reply benchmarks/kpi/results/kpi_idx_14/transformer/train.png:1]: Agreed, in the next iteration, we can fine-tune the labels.

File: benchmarks/README.md (hunk 6 above)
Comment (ab93) [reply benchmarks/README.md:7]: For sure. Slowly we can add more details here.

File: benchmarks/kpi/README.md (hunk 5 above)
Comment (ab93) [reply benchmarks/kpi/README.md:1]: Added the license

**Review by s0nicboOm — APPROVED** [review]
